cluster/ubuntu/util.sh

```diff
@@ -447,12 +447,45 @@ function kube-down {
   done
   wait
 }
-
 # Update a kubernetes cluster with latest source
 function kube-push {
-  echo "not implemented"
-}
 
+  #stop all the kube's process & etcd 
+  source "${KUBE_ROOT}/cluster/ubuntu/${KUBE_CONFIG_FILE-"config-default.sh"}"
+
+  for i in ${nodes}; do
+    echo "Cleaning on node ${i#*@}"
+    ssh -t $i 'sudo -p "[sudo] stop all process: " service etcd stop' || true
+    ssh -t $i 'rm -f /opt/bin/kube* /etc/init/kube* /etc/init.d/kube* /etc/default/kube*; rm -rf ~/kube' || true
+  done
+
+  # Run build.sh to get the latest release
+  source "${KUBE_ROOT}/cluster/ubuntu/build.sh"
```

Comment:
This doesn't obey the usage message of kube-push.sh:
```
➜  kubernetes git:(godep-cleanup) ✗ ./cluster/kube-push.sh -h
./cluster/kube-push.sh [-m|-n <node id>] <version>
  Updates Kurnetes binaries. Can be done for all components (by default), master(-m) or specified node(-n).
  If the version is not specified will try to use local binaries.
  Warning: upgrading single node is experimental
```

From https://github.com/kubernetes/kubernetes/blob/master/cluster/kube-push.sh#L31-L34

Can this implementation deploy a specific release version or a local version? Can it deploy only to the master or only to the node?